plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,655 @@
+Otherwise, the exiting transaction is determined to be non-canonical and the challenger receives `exit bond`.
+
+Note that this challenge means it’s possible for an honest user to lose `exit bond` as they might not be aware their transaction is non-canonical.
+We address this attack vector and several mitigations in detail later.
+
+<!-- TODO: Include image of canonicity "state machine" -->
+
+
+#### Piggybacking an Exit
+
+As noted earlier, it’s possible that some participants in a transaction may not be aware that the transaction is non-canonical.
+Owners of both inputs and outputs to a transaction may want to start an exit in the case that they would receive the funds from the exit.
+However, we want to avoid the gas cost of repeatedly publishing and proving statements about the same transaction.
+We therefore allow owners of inputs or outputs to a transaction to piggyback an existing exit that references the transaction.
+
+Users must piggyback an exit within the first period.
+To piggyback an exit, an input or output owner places a bond, `piggyback bond`.
+This bond is used to cover the cost of challenges that show the input or output is spent.
+A successful challenge blocks the specified input or output from exiting.
+These challenges must be presented before the end of the second period.
+
+Note that it isn’t mandatory to piggyback an exit.
+Users who choose not to piggyback an exit are choosing not to attempt a withdrawal of their funds.
+If the chain is byzantine, not piggybacking could potentially mean loss of funds.
+
+
+#### Combining with Plasma MVP Exit Protocol
+
+The MoreVP protocol can be combined with the Plasma MVP protocol in a way that simultaneously preserves the integrity of exits and minimizes gas cost. 
+Owners of outputs on the Plasma chain should be able to start an exit via either mechanism, but not both.
+Although the two protocols use different determinations for exit priority, we still need a total ordering on exits.
+Therefore, every exit, no matter the protocol used, must be included in the same priority queue for processing.
+
+
+## Alice-Bob Scenarios
+
+### Alice & Bob are honest and cooperating:
+
+1. Alice spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator begins withholding blocks while `TX1` is still in-flight.
+Neither Alice nor Bob know if the transaction has been included in a block.
+4. Someone with access to `TX1` (Alice, Bob, or otherwise) starts an exit referencing `TX1` and places `exit bond`.
+5. Alice and Bob piggyback onto the exit and each place `piggyback bond`.
+6. Alice is honest, so she hasn’t spent `UTXO1` in any transaction other than `TX1`.
+7. After period 2, Bob receives the value of `UTXO2`, Alice does not receive the value of `UTXO1`.
+All bonds are refunded.
+
```

Comment:
Why did Alice place a bond? She was never going to get anything back.

Reply:
Yeah Alice doesn't need to place a bond here. She would want to place a bond if there was another untrusted input.

Reply:
I'll change that.